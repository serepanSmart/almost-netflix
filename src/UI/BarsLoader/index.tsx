import React from 'react';
import styled, { keyframes } from 'styled-components';
import Colors from '../Theme/Colors';

export interface IBarsLoader {
  bgColor?: string;
  dataTestId?: string;
}

const Wrapper = styled.div`
  display: flex;
`;

const loading = keyframes`
  0% {
    transform: scale(1);
  }
  20% {
    transform: scale(1, 1.8);
  }
  40% {
    transform: scale(1);
  }
`;

export const Bar = styled.div<IBarsLoader>`
  display: inline-block;
  width: 4px;
  height: 18px;
  border-radius: 4px;
  margin: 0 2px;
  animation: ${loading} 1s ease-in-out infinite;
  &:nth-child(1) {
    background-color: ${({ bgColor }) => bgColor};
    animation-delay: 0;
  }
  &:nth-child(2) {
    background-color: ${({ bgColor }) => bgColor};
    animation-delay: 0.09s;
  }
  &:nth-child(3) {
    background-color: ${({ bgColor }) => bgColor};
    animation-delay: 0.18s;
  }
  &:nth-child(4) {
    background-color: ${({ bgColor }) => bgColor};
    animation-delay: 0.27s;
  }
`;

const BarsLoader: React.FC<IBarsLoader> = ({
  bgColor = Colors.White,
  dataTestId = 'bars-loader'
}) => (
  <Wrapper data-testid={dataTestId}>
    <Bar bgColor={bgColor} />
    <Bar bgColor={bgColor} />
    <Bar bgColor={bgColor} />
    <Bar bgColor={bgColor} />
  </Wrapper>
);

export default BarsLoader;
