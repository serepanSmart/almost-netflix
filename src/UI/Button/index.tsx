import React, { PropsWithChildren } from 'react';
import ButtonContainer  from './styles';

export interface IButtonProps {
  value?: string;
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  type?: 'button' | 'submit' | 'reset';
  theme?: 'primary' | 'light' | 'reject';
  opacity?: boolean;
  icon?: boolean;
}

const Button: React.FC<PropsWithChildren<IButtonProps>> = ({
  value,
  onClick,
  type,
  theme = 'primary',
  icon,
  children,
}) => {
  return (
    <ButtonContainer
      type={type}
      onClick={onClick}
      theme={theme}
      icon={icon}
    >
      {value}
      {children}
    </ButtonContainer>
  );
};

export default Button;