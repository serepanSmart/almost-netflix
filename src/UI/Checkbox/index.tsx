import React from 'react';
import {
  CheckboxContainer,
  Icon,
  HiddenCheckbox,
  StyledCheckbox,
} from './styles';

// eslint-disable-next-line max-len
interface CheckboxProps
  extends Omit<
  React.HTMLProps<HTMLInputElement>,
  'ref' | 'as' | 'type' | 'css'
  > {
  square?: boolean;
}

const Checkbox: React.FC<CheckboxProps> = ({
  className,
  checked,
  disabled,
  ...props
}) => (
  <CheckboxContainer className={className}>
    <HiddenCheckbox
      type="checkbox"
      checked={checked}
      disabled={disabled}
      {...props}
    />
    <StyledCheckbox
      data-test-checked={checked}
      checked={checked || false}
      disabled={disabled}
    >
      <Icon viewBox="0 0 24 24">
        <polyline points="20 6 9 17 4 12" />
      </Icon>
    </StyledCheckbox>
  </CheckboxContainer>
);

export default Checkbox;
export type { CheckboxProps };
