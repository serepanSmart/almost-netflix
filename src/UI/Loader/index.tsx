import React from 'react';
import LoaderContainer from './styles';

const Loader: React.FC = () => (
  <>
    <LoaderContainer>
      <div></div>
      <div></div>
    </LoaderContainer>
  </>
);

export default Loader;
