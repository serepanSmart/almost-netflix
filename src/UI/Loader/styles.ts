import Colors from '../Theme/Colors';
import styled, { keyframes } from 'styled-components';

const spin = keyframes`
  0% {
    top: 36px;
    left: 36px;
    width: 0;
    height: 0;
    opacity: 1;
  }
  100% {
    top: 0px;
    left: 0px;
    width: 72px;
    height: 72px;
    opacity: 0;
  }
`;

const LoaderContainer = styled.div`
  display: flex;
  justify-content: center;
  position: relative;
  margin: 25px auto;
  width: 80px;
  height: 80px;

  div {
    position: absolute;
    border: 4px solid ${Colors.White};
    opacity: 1;
    border-radius: 50%;
    animation: ${spin} 1s cubic-bezier(0, 0.2, 0.8, 1) infinite;
      &:nth-child(2) {
        animation-delay: -0.5s;
      }
  }
`;

export default LoaderContainer;
