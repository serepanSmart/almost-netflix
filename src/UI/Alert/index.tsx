/* eslint-disable max-len */

import React from 'react';
import { CloseButton, Message, Title, Body, Icon, Wrapper } from './styles';
import {
  X,
  XOctagonFill,
  CheckCircleFill,
  ExclamationCircleFill,
} from '@styled-icons/bootstrap';
import Colors from '../Theme/Colors';

export interface IAlert {
  type: 'error' | 'warning' | 'info' | 'success';
  title?: string | React.ReactNode;
  message?: string | React.ReactNode;
  onClose?: () => void;
}

const Alert: React.FC<IAlert> = ({
  type,
  title,
  message,
  onClose,
  ...other
}) => (
  <Wrapper
    styled={{
      type,
      onClose,
    }}
    {...other}
  >
    <Icon>
      {type === 'success' && <CheckCircleFill fill="#50dc0a" />}
      {type === 'info' && <ExclamationCircleFill fill="#1384ec" />}
      {type === 'warning' && <ExclamationCircleFill fill="#f6ad52" />}
      {type === 'error' && <XOctagonFill fill={Colors.Scarlet} />}
    </Icon>
    <Body>
      {title && <Title>{title}</Title>}
      {message && <Message>{message}</Message>}
      {onClose && (
        <CloseButton onClick={onClose} type="button">
          <X size={30} />
        </CloseButton>
      )}
    </Body>
  </Wrapper>
);

export default Alert;
