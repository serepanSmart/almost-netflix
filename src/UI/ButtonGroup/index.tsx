import React from 'react';
import { VerticalContainer, Wrapper } from './styles';

export interface IButtonGroup {
  noWrap?: boolean;
  vertical?: boolean;
  align?: string;
  spacer?: string;
  children: React.ReactNode;
}

const ButtonGroup: React.FC<IButtonGroup> = ({
  noWrap,
  vertical,
  align,
  spacer,
  children,
  ...other
}) => (
  <Wrapper
    {...other}
    styled={{
      noWrap,
      align,
      spacer,
      vertical,
    }}
  >
    {vertical ? (
      <VerticalContainer>
        {children}
      </VerticalContainer>
    ) : children}
  </Wrapper>
);

export default ButtonGroup;