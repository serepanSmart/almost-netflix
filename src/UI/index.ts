import Alert, { IAlert } from './Alert';
import BarsLoader from './BarsLoader';
import Button from './Button';
import ButtonGroup from './ButtonGroup';
import Input from './Input';
import Loader from './Loader';
import ModalWrapper from './Modal';
import SelectContainer, {
  Option,
  ISelect,
  OptionWithoutCheckbox
} from './Select';
import TabButton, { ITab } from './TabButton';
import Colors from './Theme/Colors';
import { FormGroup, InnerGroup, InnerCol } from './FormGroup';
import Checkbox from './Checkbox';
import { StyledCheckbox } from './Checkbox/styles';
import Label from './Label';
import TextArea from './TextArea';
import DatePicker from './DatePicker';

export {
  Alert,
  BarsLoader,
  Button,
  ButtonGroup,
  Input,
  Colors,
  SelectContainer as Select, OptionWithoutCheckbox,
  ModalWrapper as Modal,
  TabButton,
  Loader,
  FormGroup, InnerGroup, InnerCol,
  Checkbox, StyledCheckbox,
  Label,
  TextArea,
  DatePicker,
};
export type { ISelect, Option, ITab, IAlert };
