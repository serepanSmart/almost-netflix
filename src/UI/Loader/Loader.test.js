import { render } from '@testing-library/react';
import Loader from '.';
import { Colors } from '@/UI';

describe('Label component', () => {
  let component;
  beforeEach(() => {
    component = () => render(<Loader />);
  });

  describe('Label renders', () => {
    test('should define Label component', () => {
      expect(component()).toBeDefined();
    });

    test('should match snapshot', () => {
      expect(component()).toMatchSnapshot();
    });

    test('spins inside should have defined colors', () => {
      const { container } = component();
      const spins = container.firstChild.querySelectorAll('div');
      spins.forEach((spin) =>
        expect(spin).toHaveStyle(`border: 4px solid ${Colors.Scarlet}`),
      );
    });
  });
});
