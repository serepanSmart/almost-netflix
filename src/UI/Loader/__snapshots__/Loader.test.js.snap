// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component Label renders should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  margin: 25px auto;
  width: 80px;
  height: 80px;
}

.c0 div {
  position: absolute;
  border: 4px solid #f65261;
  opacity: 1;
  border-radius: 50%;
  -webkit-animation: dnpXZJ 1s cubic-bezier(0,0.2,0.8,1) infinite;
  animation: dnpXZJ 1s cubic-bezier(0,0.2,0.8,1) infinite;
}

.c0 div:nth-child(2) {
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div />
        <div />
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  margin: 25px auto;
  width: 80px;
  height: 80px;
}

.c0 div {
  position: absolute;
  border: 4px solid #f65261;
  opacity: 1;
  border-radius: 50%;
  -webkit-animation: dnpXZJ 1s cubic-bezier(0,0.2,0.8,1) infinite;
  animation: dnpXZJ 1s cubic-bezier(0,0.2,0.8,1) infinite;
}

.c0 div:nth-child(2) {
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

<div>
    <div
      class="c0"
    >
      <div />
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
